
Add macroes to describe the endianness of the platform used, and to
define how bytes are to be swapped accordingly.

--- tries/libexfat/platform.h.orig	2021-12-13 10:39:39.520746794 +0000
+++ tries/libexfat/platform.h	2021-12-13 10:26:16.754074364 +0000
@@ -56,6 +56,16 @@
 #define EXFAT_LITTLE_ENDIAN _LITTLE_ENDIAN
 #define EXFAT_BIG_ENDIAN _BIG_ENDIAN
 
+#elif defined(__sun) && defined (__SVR4)
+
+#include <sys/byteorder.h>
+#define exfat_bswap16(x) BSWAP_16(x)
+#define exfat_bswap32(x) BSWAP_32(x)
+#define exfat_bswap64(x) BSWAP_64(x)
+#define EXFAT_BYTE_ORDER 1
+#define EXFAT_LITTLE_ENDIAN defined(_LITTLE_ENDIAN)
+#define EXFAT_BIG_ENDIAN defined(_BIG_ENDIAN)
+
 #else 
 #error Unknown platform
 #endif
